Subject: [PATCH] git-remote-mediawiki: find correct media for a page

When remote.<remote-name>.mediaimport is set to true, git-remote-mediawiki
attempts to fetch all images referenced from a page and include them in
each revision it commits.

It calls the function get_mw_mediafile_for_page_revision with the name
of the file and the timestamp of the page revision and expects to get
back the appropriate version of the file for that revision of the page.

It uses the MediaWiki API imageinfo query to return the file's information
passing in the timestamp of the page's revision.

However, it sets both iistart and iiend to the page revision's timestamp.
This requires the media file to have exactly the same timestamp of the
page in order for it to be found.  This is incorrect.

The media file may actually have a timestamp earlier than the page.

Fix the problem by omitting the iiend timestamp which will cause the API
to return information in newest to oldest order starting with the first item
that has a timestamp equal to or before the iistart value (which is the
passed in timestamp of the page's revision).  Since the number of information
records to be returned is limited to 1 (by the iilimit value), that will
always result in the correct image version for the page even when the image's
timestamp precedes the page's.

Signed-off-by: Kyle J. McKay <mackyle@gmail.com>

---
 contrib/mw-to-git/git-remote-mediawiki.perl | 1 -
 1 file changed, 1 deletion(-)

diff --git a/contrib/mw-to-git/git-remote-mediawiki.perl b/contrib/mw-to-git/git-remote-mediawiki.perl
index cbba8f39..caa9abcd 100755
--- a/contrib/mw-to-git/git-remote-mediawiki.perl
+++ b/contrib/mw-to-git/git-remote-mediawiki.perl
@@ -434,7 +434,6 @@ sub get_mw_mediafile_for_page_revision {
 		prop => 'imageinfo',
 		titles => "File:${filename}",
 		iistart => $timestamp,
-		iiend => $timestamp,
 		iiprop => 'timestamp|archivename|url',
 		iilimit => 1
 	};
---
